JASP-Engine/JASP/R/ancova.R

```diff
@@ -308,14 +308,18 @@ Ancova <- function(jaspResults, dataset = NULL, options) {
                  allowEmptyDataset = FALSE,
                  exitAnalysisIfErrors = TRUE,
                  custom = function() {
-                   if (grepl(tryContrMat[1], pattern = "singular contrast matrix")) {
-                     return("Singular custom contrast matrix.")
-                   } else if (grepl(tryContrMat[1], pattern = "number of contrast matrix rows")) {
-                     return("Wrong number of custom contrast matrix rows.")
+                   if (isTryError(tryContrMat)) {
+                     if (grepl(tryContrMat[1], pattern = "singular contrast matrix")) {
+                       return("Singular custom contrast matrix.")
+                     } else if (grepl(tryContrMat[1], pattern = "number of contrast matrix rows")) {
+                       return("Wrong number of custom contrast matrix rows.")
+                     } 
+                   } else if (is.matrix(tryContrMat) & any(apply(tryContrMat, 2, function(x) all(x == 0) ))) {
+                     return("Please specify non-zero contrast weights.")
                    } else if (ncol(tryContrMat) >= nlevels(dataset[[v]])) {
                      return("Please specify fewer contrasts. (Maximum #contrasts = #levels - 1)")
-                   } else if (any(round(colSums(tryContrMat), 15) != 0)) {
-                     return("Some contrasts do not sum to 0.")
+                   # } else if (any(round(colSums(tryContrMat), 15) != 0)) {
```

Comment:
Why is this commented out? Right now I can do contrasts like this but that does not seem like a legit contrast :D
![Screen Shot 2020-02-06 at 10 46 47](https://user-images.githubusercontent.com/7093546/73925272-1c17b500-48ce-11ea-9284-ffcdbc9c66b0.png)

Reply:
Yeah, at first I added the restriction that the contrasts need to sum to 0, but apparently it can be legit to not have it like that (setting one level to 0 and one to 1, you would get the marginal mean for instance). I  commented it out because we might reinstate it, but as long as R is not complaining I think we can leave it like this.